src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreResult.cs

```diff
@@ -107,62 +121,56 @@ public ISet<LibraryRange> GetAllUnresolved()
-        {
-            await CommitAsync(log, forceWrite: false, token: token);
-        }
-
-        /// <summary>
-        /// Commits the lock file contained in <see cref="LockFile"/> and the MSBuild targets/props to
-        /// the local file system.
-        /// </summary>
-        /// <remarks>If <see cref="PreviousLockFile"/> and <see cref="LockFile"/> are identical
-        ///  the file will not be written to disk.</remarks>
-        /// <param name="forceWrite">Write out the lock file even if no changes exist.</param>
-        public async Task CommitAsync(ILogger log, bool forceWrite, CancellationToken token)
+        public virtual async Task CommitAsync(ILogger log, CancellationToken token)
         {
             // Write the lock file
             var lockFileFormat = new LockFileFormat();
 
             var isTool = ProjectStyle == ProjectStyle.DotnetCliTool;
 
             // Commit the assets file to disk.
-            await CommitAsync(
+            await CommitAssetsFileAsync(
                 lockFileFormat,
                 result: this,
                 log: log,
-                forceWrite: forceWrite,
                 toolCommit: isTool,
                 token: token);
+            
+            //Commit the cache file to disk
+            await CommitCacheFileAsync(
+                log: log,
+                toolCommit : isTool);
         }
 
-        private static async Task CommitAsync(
+        private async Task CommitAssetsFileAsync(
             LockFileFormat lockFileFormat,
             IRestoreResult result,
             ILogger log,
-            bool forceWrite,
             bool toolCommit,
             CancellationToken token)
         {
             // Commit targets/props to disk before the assets file.
             // Visual Studio typically watches the assets file for changes
             // and begins a reload when that file changes.
             var buildFilesToWrite = result.MSBuildOutputFiles
-                    .Where(e => forceWrite || BuildAssetsUtils.HasChanges(e.Content, e.Path, log));
+                    .Where(e => BuildAssetsUtils.HasChanges(e.Content, e.Path, log));
 
             BuildAssetsUtils.WriteFiles(buildFilesToWrite, log);
 
             // Avoid writing out the lock file if it is the same to avoid triggering an intellisense
             // update on a restore with no actual changes.
-            if (forceWrite
-                || result.PreviousLockFile == null
+            if (result.PreviousLockFile == null
                 || !result.PreviousLockFile.Equals(result.LockFile))
             {
                 if (toolCommit)
                 {
                     if (result.LockFilePath != null && result.LockFile != null)
                     {
-                        log.LogDebug($"Writing tool lock file to disk. Path: {result.LockFilePath}");
+                        
+
+                        log.LogDebug(string.Format(CultureInfo.CurrentCulture,
```

Comment:
debug is typically non-localized, but I don't think it is a problem to localize it either here